File: cpp/Video 05-02 (JAVA CS)Composition Aggregation Assoication/Java/Aggregation/Computer.java
class Computer
{

	private  Processor processor;
	private String operatingSystem;
	public void setOperatingSystem(String operatingSystem){
		this.operatingSystem = operatingSystem;
	}
	String getOperatingSystem(){
		return this.operatingSystem;
	}
	Computer(){
                 
		 
	}
	 
	Processor getProcessor()
	{
		
			return this.processor;
		
	}
	
	void setProcessor(Processor processor)
	{
			this.processor = processor;
			 
			
			 
			  
	}
	
    void print(){
	  System.out.println("Operating System"+operatingSystem);
      System.out.println("Clock Speed "+processor.getClockSpeed());
	}	
}

 



	
 

File: cpp/Video 05-02 (JAVA CS)Composition Aggregation Assoication/CS/OOP/Composition/Computer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Association
{
    class Computer
    {
        private Processor processor;
        private String operatingSystem;

        public string OperatingSystem
        {
            get { return operatingSystem; }
            set { operatingSystem = value; }
        }

        public Processor Processor
        {
            get { return processor; }
            set
            {
                processor = value;
                processor.ClockSpeed = 5.0;
            }
        }
        public void Print()
        {
            
            Console.WriteLine("Operating System" + OperatingSystem);
            Console.WriteLine("Clock Speed " + processor.ClockSpeed);
        }
        
    }
}
